pkg/vpcmodel/nodesConnectivity.go

```diff
@@ -78,6 +78,29 @@ func (c *VPCConfig) getLoadBalancerRule(src, dst Node) LoadBalancerRule {
 	return nil
 }
 
+func (c *VPCConfig) getPrivateSubnetRule(src, dst Node) PrivateSubnetRule {
+	switch {
+	case dst.IsInternal():
```

Comment:
Wouldn't it be more efficient and also more readable to check here External instead of Internal?
(If both are internal instead of returning nil directly you call `GetPrivateSubnetRule` which return nil in this case)

Reply:
the creating a rule is done inside the subnet, the subnet is the struct that "know" the logic of creating the rule.
this code is in the node-connectivity calculation. it should not know anything about the logic of creating the rule.
this switch is only for capturing the subnet.
in ideal world (which has no golang)   GetPrivateSubnetRule() would be a static function of the subnet struct. and this switch was not necessary.